src/Core/Checkout/Cart/Rule/AffiliateCodeOfOrderRule.php

```diff
@@ -39,7 +42,13 @@ public function match(RuleScope $scope): bool
             return RuleComparison::isNegativeOperator($this->operator);
         }
 
-        return RuleComparison::string($affiliateCode, $this->affiliateCode ?? '', $this->operator);
+        return RuleComparison::stringArray(
+            $affiliateCode,
+            $this->affiliateCode !== null ? array_values(
```

Comment:
Might be worth putting this code in the `stringArray` static constructor?

Reply:
Yes probably. In general I was also not sure, why in the `RuleComparison::stringArray` we have a `mb_lowercase` conversion. Maybe it would also be more reasonable, to add this to another parameter and do not do the conversion at all? 

Besides that I noticed that in the other place we do not use the multibyte version of that: https://github.com/shopware/shopware/blob/3dfe23d832f0459e0903004550620baf384e893f/src/Core/Checkout/Customer/Rule/CustomerNumberRule.php#L45
Which should probably be fine, but one never knows what the Shop admins put in their customer numbers :D